src/main/java/book/store/onlinebookstore/model/ShoppingCart.java

```diff
@@ -1,4 +1,49 @@
 package book.store.onlinebookstore.model;
 
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.MapsId;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@Entity
+@Table(name = "shopping_carts")
+@Data
 public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @MapsId
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @JoinColumn(nullable = false, name = "user_id")
+    @OneToOne(fetch = FetchType.LAZY, optional = false)
+    private User user;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @OneToMany(mappedBy = "shoppingCart", orphanRemoval = true, cascade = CascadeType.REMOVE)

```

Comment:
I would change cascade to `ALL` since cart item lifecycle is completely bound to shopping cart